### PR TITLE
[TRITON][GLUON] Prefill MQA Logits kernel tuning for GLM 5.x

### PR DESCRIPTION
**Pull request by cagrikymk**
## Motivation

GLM 5 series use number of q-heads = 32, but this kernel was originally tuned for 64 heads. 

This PR improves performance for GLM 5 via kernel updates + tuning.

## Technical Details

Now the kernel supports handling 2 tokens together to increase the work per WG (since 32*2=64). While there is still some performance gap between GLM 5 config and dsv4 one, this PR closes that gap quite a bit.

## Test Plan

Existing tests are extended to include GLM 5 case as well.

## Test Result

Tests pass.

## Performance

While DSV4 config (num_heads=64) shows some regression when # tokens~= 4k, specifically 1k/1k case, those cases appear rarely during inference as batching typically goes for >= 8k tokens. 

Results are from MI355.

Baseline and new column format: `{time} us / TFLOPs`
## num_heads = 32

| ISL=OSL | batch | rows | baseline | new | speedup |
|---|---|---|---|---|---|
| 1024 | 1 | 1,024 | 15.0 µs / 287 | 10.7 µs / 401 | **1.40x** |
| 1024 | 4 | 4,096 | 27.2 µs / 633 | 27.6 µs / 624 | 0.99x |
| 1024 | 8 | 8,192 | 50.4 µs / 682 | 39.1 µs / 879 | **1.29x** |
| 1024 | 16 | 16,384 | 93.3 µs / 738 | 71.3 µs / 965 | **1.31x** |
| 2048 | 1 | 2,048 | 31.8 µs / 541 | 25.4 µs / 678 | **1.25x** |
| 2048 | 2 | 4,096 | 46.3 µs / 743 | 44.8 µs / 767 | **1.03x** |
| 2048 | 4 | 8,192 | 85.6 µs / 803 | 63.0 µs / 1,091 | **1.36x** |
| 2048 | 8 | 16,384 | 162.7 µs / 845 | 117.8 µs / 1,167 | **1.38x** |
| 4096 | 1 | 4,096 | 83.2 µs / 827 | 81.2 µs / 847 | **1.02x** |
| 4096 | 2 | 8,192 | 152.5 µs / 902 | 106.5 µs / 1,291 | **1.43x** |
| 4096 | 4 | 16,384 | 293.3 µs / 937 | 200.2 µs / 1,373 | **1.47x** |
| 8192 | 1 | 8,192 | 288.4 µs / 953 | 194.1 µs / 1,417 | **1.49x** |
| 8192 | 2 | 16,384 | 571.7 µs / 962 | 376.6 µs / 1,460 | **1.52x** |
| | | **geomean** | | | **1.314x** |

## num_heads = 64

| ISL=OSL | batch | rows | baseline | new | speedup |
|---|---|---|---|---|---|
| 1024 | 1 | 1,024 | 18.9 µs / 456 | 18.6 µs / 462 | 1.02x |
| 1024 | 4 | 4,096 | 34.0 µs / 1,012 | 40.5 µs / 850 | 0.84x |
| 1024 | 8 | 8,192 | 61.2 µs / 1,123 | 62.6 µs / 1,099 | 0.98x |
| 1024 | 16 | 16,384 | 115.6 µs / 1,190 | 110.5 µs / 1,245 | **1.05x** |
| 2048 | 1 | 2,048 | 39.3 µs / 876 | 38.9 µs / 885 | 1.01x |
| 2048 | 2 | 4,096 | 56.5 µs / 1,217 | 59.5 µs / 1,155 | 0.95x |
| 2048 | 4 | 8,192 | 104.8 µs / 1,312 | 100.9 µs / 1,363 | **1.04x** |
| 2048 | 8 | 16,384 | 193.4 µs / 1,422 | 187.7 µs / 1,465 | **1.03x** |
| 4096 | 1 | 4,096 | 99.6 µs / 1,380 | 97.6 µs / 1,408 | **1.02x** |
| 4096 | 2 | 8,192 | 178.3 µs / 1,542 | 178.6 µs / 1,539 | 1.00x |
| 4096 | 4 | 16,384 | 340.1 µs / 1,617 | 329.0 µs / 1,672 | **1.03x** |
| 8192 | 1 | 8,192 | 330.9 µs / 1,662 | 313.7 µs / 1,753 | **1.05x** |
| 8192 | 2 | 16,384 | 651.6 µs / 1,688 | 613.9 µs / 1,791 | **1.06x** |
| | | **geomean** | | | **1.009x** |

### CODE REVIEW

File: aiter/ops/triton/_gluon_kernels/gfx950/attention/fp8_mqa_logits.py [hunk 1]
```diff
@@ -135,6 +135,12 @@ def _make_kv_load_layouts_cdna4(
             order=[0, 1],
         )
         shared = gl.SwizzledSharedLayout(vec=1, per_phase=1, max_phase=1, order=[0, 1])
+        # shared = gl.SwizzledSharedLayout(
```
Comment (vgokhale) [inline aiter/ops/triton/_gluon_kernels/gfx950/attention/fp8_mqa_logits.py:140]: Delete?

**Review by vgokhale — APPROVED** [review]

File: aiter/ops/triton/_gluon_kernels/gfx950/attention/fp8_mqa_logits.py (hunk 1 above)
Comment (cagrikymk) [reply aiter/ops/triton/_gluon_kernels/gfx950/attention/fp8_mqa_logits.py:140]: Swizzling reduces LDS conflicts drastically but was slightly hurting perf., potentially related to the high VALU/MFMA ratio. So, I want to keep that around in case the kernel starts getting bottlenecked by LDS BW.
